azkaban-exec-server/src/main/java/azkaban/execapp/JobRunner.java

```diff
@@ -694,6 +699,12 @@ private void doRun() {
         writeStatus();
         fireEvent(Event.create(this, EventType.JOB_STATUS_CHANGED,
             new EventData(Status.RUNNING, this.node.getNestedId())));
+        try {
+          addGroupAndUser(this.jobtypeManager.getCommonPluginLoadProps());
```

Comment:
Wondering if this logic classifies under "job preparation" and therefore doing it inside `prepareJob()` would be more appropriate/expected.

Reply:
prepareJob is more like setting all the necessary properties. That is why I kept it out. Please let me know if you strongly feel about it. I will move it.